rocAL/rocAL/source/pipeline/ring_buffer.cpp

```diff
@@ -209,11 +209,12 @@ void RingBuffer::init(RocalMemType mem_type, void *devres, std::vector<size_t> &
 
 void RingBuffer::initBoxEncoderMetaData(RocalMemType mem_type, size_t encoded_bbox_size, size_t encoded_labels_size)
 {
+    _box_encoder = true;
 #if ENABLE_HIP
     DeviceResourcesHip *dev_hip = static_cast<DeviceResourcesHip *>(_dev);
     if(_mem_type == RocalMemType::HIP)
     {
-        _box_encoder_gpu = true;
+        _box_encoder = true;
```

Comment:
We are setting to true in beginning itself then why are we setting again in Line number 217 and 241